src/sidebar/components/OpenDashboardMenuItem.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'preact/hooks';
+
+import { withServices } from '../service-context';
+import type { DashboardService } from '../services/dashboard';
+import MenuItem from './MenuItem';
+
+export type OpenDashboardMenuItemProps = {
+  isMenuOpen: boolean;
+
+  // Injected
+  dashboard: DashboardService;
+};
+
+function OpenDashboardMenuItem({
+  dashboard,
+  isMenuOpen,
+}: OpenDashboardMenuItemProps) {
+  const [authToken, setAuthToken] = useState<string>();
+  useEffect(() => {
+    // Fetch a new auth token every time the menu containing this item is open,
+    // to make sure we always have an up-to-date one
+    if (isMenuOpen) {
+      dashboard.getAuthToken().then(setAuthToken);
+    }
+
+    // Discard previous token just before trying to fetch a new one
+    return () => setAuthToken(undefined);
+  }, [dashboard, isMenuOpen]);
+
+  return (
+    <MenuItem
+      label="Open dashboard"
+      isDisabled={!authToken}
+      onClick={() => authToken && dashboard.open(authToken)}
```

Comment:
There's what seems to be an inconsistency in `MenuItem` which makes it still render with cursor pointer, even if the item is disabled.

Also, the prop JS doc mentiones the `onClick` callback is still invoked, which should be fine here because `authToken` is checked before invoking `dashboard.open(authToken)`.

I will address these inconsistencies afterwards on a follow-up PR.

Reply:
See https://github.com/hypothesis/client/pull/6325